training/src/app/chuck-noris/search/search.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { debounceTime, distinctUntilChanged, Subject, takeUntil, tap } from "rxjs";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss']
+})
+export class SearchComponent implements OnInit {
+	public searchForm: FormGroup;
+	@Output() public generateJokesFromSearch: EventEmitter<string> = new EventEmitter<string>();
+	@Output() public generateJokesFromClick: EventEmitter<string>  = new EventEmitter<string>();
+	public shouldDestroy: Subject<void>                            = new Subject<void>();
```

Comment:
we useally name boolean values with "should..." "isValid" "hasSomething" etc.  this is not a boolean its a subject